src/Psy/TabCompletion/Matchers/AbstractMatcher.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Psy\TabCompletion\Matchers;
+
+use Psy\Context;
+use Psy\ContextAware;
+
+/**
+ * Class AbstractMatcher
+ * @package Psy\TabCompletion\Matchers
+ */
+abstract class AbstractMatcher implements ContextAware
```

Comment:
It looks like only three matchers actually need ContextAware (ObjectAttributes, ObjectMethods, and Variables). Let's extend this with a ContextAwareMatcher (or AbstractContextAwareMatcher, I don't care which name) for them to extend so that only those three matchers are actually ContextAware.